Documentation/teaching/labs/block_device_drivers.rst

```diff
@@ -0,0 +1,1160 @@
+====================
+Block Device Drivers
+====================
+
+Lab objectives
+==============
+
+  * acquiring knowledge about the behavior of the I/O subsystem on Linux
+  * hands-on activities in structures and functions of block devices
+  * acquiring basic skills for utilizing the API for block devices, by solving
+    exercises
+
+Overview
+========
+
+Block devices are characterized by random access to data organized in fixed-size
+blocks. Examples of such devices are hard drives, CD-ROM drives, RAM disks, etc.
+The speed of block devices is generally much higher than the speed of character
+devices, and their performance is also important. This is why the Linux kernel
+handles differently these 2 types of devices (it uses a specialized API).
+
+Working with block devices is therefore more complicated than working with
+character devices. Character devices have a single current position, while block
+devices must be able to move to any position in the device to provide random
+access to data. To simplify work with block devices, the Linux kernel provides
+an entire subsystem called the block I/O (or block layer) subsystem.
+
+From the kernel perspective, the smallest logical unit of addressing is the
+block. Although the physical device can be addressed at sector level, the kernel
+performs all disk operations using blocks. Since the smallest unit of physical
+addressing is the sector, the size of the block must be a multiple of the size
+of the sector. Additionally, the block size must be a power of 2 and can not
+exceed the size of a page. The size of the block may vary depending on the file
+system used, the most common values being 512 bytes, 1 kilobytes and 4
+kilobytes.
+
+Block I/O devices in Linux
```

Comment:
I don't think that we need this section, please remove it and bump up all of the subsections.